exercises/practice/nucleotide-count/.meta/Example.roc

```diff
@@ -0,0 +1,20 @@
+module [nucleotideCounts]
+
+nucleotideCounts : Str -> Result { a : U64, c : U64, g : U64, t : U64 } _
+nucleotideCounts = \input ->
+    bytes = Str.toUtf8 input
+    containsInvalidNucleotide = List.any bytes \byte ->
+        List.contains ['A', 'C', 'G', 'T'] byte |> Bool.not
+
+    if containsInvalidNucleotide then
+        Err InvalidNucleotide
+        else
+
+    List.walk bytes { a: 0, c: 0, g: 0, t: 0 } \state, elem ->
+        when elem is
+            'A' -> { state & a: state.a + 1 }
+            'C' -> { state & c: state.c + 1 }
+            'G' -> { state & g: state.g + 1 }
+            'T' -> { state & t: state.t + 1 }
+            _ -> crash "impossible"
+    |> Ok
```

Comment:
To avoid going through the bytes twice, perhaps instead of `_ -> crash "Impossible"` we could return `Break (Err InvalidNucleotide)` and use `List.walkUntil` instead of `List.walk`? Just an idea, not important at all.

Reply:
Ah yeah great idea! I will update this to use List.walkTry which will be a bit simpler than walkUntil.